src/common/contexts/provider.tsx

```diff
@@ -95,17 +194,81 @@ export const ProviderContextProvider: FunctionComponent<ProviderContextProviderP
     return initialiseMagicSigner();
   };
 
+  const reloadNetwork = async () => {
+    const provider = getProvider();
+    if (!provider) throw new UnsupportedNetworkError();
+
+    const chainId = (await provider.getNetwork()).chainId;
+    await changeNetwork(chainId);
+  };
+
   useEffect(() => {
-    // Do not listen before the provider is upgraded by the app
-    if (providerType !== SIGNER_TYPE.METAMASK) return;
-    window.ethereum.on("accountsChanged", () => {
-      return initializeMetaMaskSigner();
-    });
-  }, [providerType]);
+    currentProvider = getProvider();
+    (async () => {
+      const provider = getProvider();
+      if (!provider) {
+        setCurrentChainId(undefined);
+      } else {
+        const network = await provider.getNetwork();
+        setCurrentChainId(network.chainId);
+      }
+    })();
```

Comment:
need to ask the rest whether they are comfortable with seeing IIFEs, ow better to just use a named lambda and call it immediately (yea less elegant)

Reply:
Sure, we can also change to the normal way if we are not comfortable with IIFE. Let me know.👍

Reply:
Updated the IIFE to normal functions.